### PR TITLE
#5233: run convs with auto-format

### PR DESCRIPTION
**Pull request by arakhmati**
This PR grew a bit :eyes:

- Added assert for making sure that ttnn ops are registered with unique names
- Added support for running `torchvision.models.resnet.BasicBlock` without modifications
- Added `ttnn.query_all_registered_operations()`
- Renamed logical_not_unary to logical_unary

### CODE REVIEW

File: tests/ttnn/integration_tests/resnet/test_ttnn_functional_resnet.py [hunk 1]
```diff
@@ -276,23 +296,35 @@ def custom_preprocessor(model, name, ttnn_module_args, convert_to_ttnn):
...
+    output_tensor = ttnn.to_layout(conv2, ttnn.ROW_MAJOR_LAYOUT)
+    output_tensor = ttnn.reshape(output_tensor, (8, 1, 49, 512))
+    output_tensor = ttnn.to_layout(output_tensor, ttnn.TILE_LAYOUT)
+    output_tensor = ttnn.global_avg_pool2d(output_tensor)
     output_tensor = output_tensor @ parameters.fc.weight + parameters.fc.bias
 
     output_tensor = ttnn.to_torch(output_tensor)
+    output_tensor = torch.reshape(output_tensor, (8, 1000))
+
+    # assert_with_pcc(torch_output_tensor, output_tensor, pcc=0.999)
```
Comment (cfjchu) [inline tests/ttnn/integration_tests/resnet/test_ttnn_functional_resnet.py:330]: was this supposed to be uncommented?

**Review by cfjchu — APPROVED** [review]

File: tests/ttnn/integration_tests/resnet/test_ttnn_functional_resnet.py [hunk 2]
```diff
@@ -22,102 +24,43 @@
...
+    ttnn_module_args["conv_blocking_and_parallelization_config_override"] = {"act_block_h": 32}
+
+
 def custom_preprocessor(model, name, ttnn_module_args):
     parameters = {}
-    if isinstance(model, BasicBlock):
+    if isinstance(model, torchvision.models.resnet.BasicBlock):
         ttnn_module_args.conv1["activation"] = "relu"  # Fuse relu with conv1
 
         conv1_weight, conv1_bias = fold_batch_norm2d_into_conv2d(model.conv1, model.bn1)
```
Comment (eyonland) [inline tests/ttnn/integration_tests/resnet/test_ttnn_functional_resnet.py:38]: I'm realizing that we don't have much documentation around the model_preprocessing.py in general.    For example fold_batch_norm2d_into_conv2d.   Do we need to start adding doc here?  Do we want everyone to feel free to just add functions like this to model_preprocessing.py going forward?

**Review by eyonland — APPROVED** [review]

**Review by tt-aho — APPROVED** [review]

File: models/experimental/functional_resnet/tt/ttnn_functional_resnet.py [hunk 3]
```diff
@@ -22,16 +22,21 @@ def __call__(self, x):
         identity = x
 
         # Relu and bn1 are fused with conv1
-        out = self.conv1(x)
+        conv1 = self.conv1(x)
 
         # Relu and bn2 are fused with conv1
-        out = self.conv2(out)
+        conv2 = self.conv2(conv1)
```
[the remark this answers is not in the document]
Comment (arakhmati) [reply models/experimental/functional_resnet/tt/ttnn_functional_resnet.py:28]: I needed it to be able to call `ttnn.deallocate`

File: tests/ttnn/integration_tests/resnet/test_ttnn_functional_resnet.py (hunk 2 above)
Comment (arakhmati) [reply tests/ttnn/integration_tests/resnet/test_ttnn_functional_resnet.py:38]: We need to document our use cases. And the process will be the same as always. People need to write their versions and if they are general enough, then we merge them into model_preprocessing.py

**Pull request merged**